src/libraries/Common/src/Interop/FreeBSD/Interop.Process.GetProcInfo.cs

```diff
@@ -224,6 +225,11 @@ public unsafe struct kinfo_proc
 
                 count = (int)bytesLength / sizeof(kinfo_proc);
             }
+            catch (InvalidOperationException)
+            {
+                // Sysctl() has freed the pBuffer which it allocated for us.
+                throw;
+            }
```

Comment:
Won't pBuffer have been set to null if the method throws InvalidOperationException?

Reply:
It seems like we are not setting pBuffer to zero in the exceptional case in `Sysctl`: https://github.com/dotnet/runtime/blob/b6af738eabf08b7e450e8a73290c2baca8635334/src/libraries/Common/src/Interop/BSD/System.Native/Interop.Sysctl.cs#L56-L57 and underlying `FreeHGlobal` on Unix just calls libc `free` (just makes the pointer dangling but the address value stays). Currently, we appear to be double freeing the memory in case of `InvalidOperationException`.

Reply:
The value parameter is set to null at the beginning of the method and isn't set back to pBuffer until after these throws.  Am I reading it wrong? Those checks for value == null look pointless.

Reply:
You are right. I removed the temp `pBuffer` from `Sysctl()` helper and added a check in the `finally` block to simplify the flow.